data/base-sorting.yaml

```diff
@@ -72,7 +72,7 @@ jewelry:
 - pendant
 - pin
 - ring
-- star
+- (?<!morning) star
```

Comment:
Does the space between morning and star need to be in the right parenthesis to avoid matching " star" instead of "star"?

```suggestion
- (?<!morning )star
```

I haven't tested, just asking

Reply:
That was my first version too, and the regex match fails that way. I tried `(?<!morning\s)star` and `(?<!morning )star` And neither matched morning star. This way works for morning star, and I don't have a plain star to test. My original thought was to delete `star` altogether. :) Whenif I ever get a `star` I'll test it though.

Reply:
> Does the space between morning and star need to be in the right parenthesis to avoid matching " star" instead of "star"?
> 
> I haven't tested, just asking

Need moar Katoak!

Reply:
I'm confused - is the goal to match `morning star` or not morning star?

Reply:
I want to match morning star. There is an entry for "morning star" under weapons. This works. 

There is an entry for plain "star" under jewelry. I want this entry to only match "star" without "morning " in front of it. The latter I can't seem to make happen.

Reply:
`^(?!morning )star` seems to work for me:

* morning star = no match
* star = match

https://rubular.com/r/9D7IFASCbS8h7q

Reply:
Of course. I tell @rcuhljr this all the time... you know stuff I can't conceive of. 

I use the syntax I'm trying in profanity and it didn't occur to me there was another.

Reply:
Thanks for that link too. Looks super useful!

Reply:
So, this doesn't work either. 
![Screenshot from 2022-04-13 10-44-11](https://user-images.githubusercontent.com/93822896/163066678-0c027861-7216-452c-b4d9-80af5781fd3a.png)

```
vimes:~/lich/scripts/data$ egrep morning base-sorting.yaml 
- ^(?!morning )star
```

I'm inclined to just remove `star` :)

Reply:
So, clearly the parsing has some issue. I don't want to tackle that right now, but I do want my morning star to match correctly.

How about we compromise on changing `star` to `stars` which will accomplish what I want, and looking at `https://drservice.info/Plaza/` suggests would also catch most of the jewelry correctly.

Reply:
https://rubular.com/r/9D7IFASCbS8h7q says the lookahead I coded up front should work too, but it doesn't.

Any which way I've tried, if I have the entry under `jewelry` it seems to overwrite the `morning star` entry and the morning star gets categorised under jewelry.

Reply:
Still doesn't work with the prefix. This is driving me mad. We have other regex in there, and those seem to work just fine.

Reply:
the entry under weapons is a regex, note, and that works (If I comment out the entry under jewelry) `(morning|throwing) star` works. We have all kinds of positive lookaheads in here. Seems just the negative doesn't work?

Reply:
Oh my goodness, it works now! `/(?<!morning )star/` was the magic incantation.

Reply:
Bah, declared victory too early. Even with `/(?<!morning )star/` if I comment out the weapon line, my morning star registers as jewelry. :( I'm sad.

Reply:
Hah, disregard. I was a tad careless. With `/(?<!morning )star/` it does work!  It had the morning star under `other` which I just noticed when I double checked.